platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/GossipRpcShadowgraphSynchronizer.java

```diff
@@ -0,0 +1,446 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.swirlds.platform.gossip.shadowgraph;
+
+import static com.swirlds.platform.gossip.shadowgraph.SyncUtils.getMyTipsTheyKnow;
+import static com.swirlds.platform.gossip.shadowgraph.SyncUtils.getTheirTipsIHave;
+import static org.hiero.base.CompareTo.isGreaterThanOrEqualTo;
+
+import com.hedera.hapi.platform.event.GossipEvent;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.logging.legacy.LogMarker;
+import com.swirlds.platform.gossip.IntakeEventCounter;
+import com.swirlds.platform.gossip.rpc.GossipRpcReceiver;
+import com.swirlds.platform.gossip.rpc.GossipRpcSender;
+import com.swirlds.platform.gossip.rpc.SyncData;
+import com.swirlds.platform.gossip.sync.config.SyncConfig;
+import com.swirlds.platform.metrics.SyncMetrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hiero.base.crypto.Hash;
+import org.hiero.consensus.gossip.FallenBehindManager;
+import org.hiero.consensus.model.event.PlatformEvent;
+import org.hiero.consensus.model.node.NodeId;
+
+/**
+ * The goal of the GossipRpcShadowgraphSynchronizer is to compare graphs with a remote node, and update them so both
+ * sides have the same events in the graph. This process is called a sync.
+ * <p>
+ * This class is designed to use message based protocol (as a precursor to RPC communication) instead of stealing entire
+ * socket (unlike {@link ShadowgraphSynchronizer}
+ */
+public class GossipRpcShadowgraphSynchronizer extends AbstractShadowgraphSynchronizer {
```

Comment:
I think this class should be broken into smaller pieces. It has too many responsibilities.